In [1]:
library(tidyverse)
library(dbplyr)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors

Attaching package: ‘dbplyr’


The following objects are masked from ‘package:dplyr’:

    ident, sql




In [2]:
raw_data <- read.csv("data/full_data.csv", header = TRUE)


In [3]:
head(raw_data)

,cohort,person_id,DOB,Gender,Ethnicity,EAL,ever_FSM,ever_CIN,DistCurrSch,DistNearSch,⋯,Y2015.2016_Term1,Y2015.2016_Term2,Y2015.2016_Term3,Y2016.2017_Term1,Y2016.2017_Term2,Y2016.2017_Term3,Y2017.2018_Term1,Y2017.2018_Term2,Y2017.2018_Term3,ever_AP
,<chr>,<chr>,<chr>,<chr>,<chr>,<int>,<int>,<lgl>,<dbl>,<dbl>,⋯,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<lgl>
1,2000/01,0002F12C1F92EC01CB0C05D441AA6D7AE16E9CD131BB64885744D08BA74E1FDD,2001-05-15,M,White British,0,0,FALSE,0.66,0.66,⋯,107442,107442,107442,143114,143114,143114,NA,NA,NA,FALSE
2,2000/01,00034E776A98176C2B6075634BC4A614400F32AA9C4D599FDDE156541F1F9A29,2001-01-15,F,White British,0,0,TRUE,3.50,0.62,⋯,140569,140569,140569,140569,140569,140569,NA,NA,NA,FALSE
3,2001/02,0004E60FEFC16781A48973C80F782B233869629C9CE44B741E2D2043047FCB1D,2001-11-15,M,White British,0,1,TRUE,7.01,0.00,⋯,136529,136529,136529,136529,136529,136529,136529,136529,136529,FALSE
4,2001/02,0013020C6ED2FF78BCA8E5F2D0B9B16E66C50B6A2AABB754FB63314CFE3A2EC7,2002-05-15,M,Other,1,0,FALSE,0.57,0.57,⋯,107441,107441,107441,143112,143112,143112,143112,143112,143112,FALSE
5,2000/01,00149D26484ED3BE2DEF154B8DB53A36B1A9FF0839C447A2CE19B0750225BA66,2000-11-15,M,South Asian,1,0,FALSE,1.69,0.44,⋯,108087,108087,108087,108087,108087,108087,NA,NA,NA,FALSE
6,2000/01,0014DF7EADF7C22A0FD2F6862B6F06778B199F1F2D61242125DA8EC539E6275D,2001-04-15,F,White British,0,0,FALSE,1.12,0.62,⋯,107366,107366,107366,142761,142761,142761,NA,NA,NA,FALSE


# Split cohorts, rename to Year7-11 and rejoin

In [4]:
cohort1 <- raw_data |>
    filter(cohort == '2000/01') |>
    select(-c(Y2017.2018_Term1, Y2017.2018_Term2, Y2017.2018_Term3)) |>
    rename(
        Year7_Term1 = Y2012.2013_Term1,
        Year7_Term2 = Y2012.2013_Term2,
        Year7_Term3 = Y2012.2013_Term3,
        Year8_Term1 = Y2013.2014_Term1,
        Year8_Term2 = Y2013.2014_Term2,
        Year8_Term3 = Y2013.2014_Term3,
        Year9_Term1 = Y2014.2015_Term1,
        Year9_Term2 = Y2014.2015_Term2,
        Year9_Term3 = Y2014.2015_Term3,
        Year10_Term1 = Y2015.2016_Term1,
        Year10_Term2 = Y2015.2016_Term2,
        Year10_Term3 = Y2015.2016_Term3,
        Year11_Term1 = Y2016.2017_Term1,
        Year11_Term2 = Y2016.2017_Term2,
        Year11_Term3 = Y2016.2017_Term3
        )

In [5]:
head(cohort1)

,cohort,person_id,DOB,Gender,Ethnicity,EAL,ever_FSM,ever_CIN,DistCurrSch,DistNearSch,⋯,Year9_Term1,Year9_Term2,Year9_Term3,Year10_Term1,Year10_Term2,Year10_Term3,Year11_Term1,Year11_Term2,Year11_Term3,ever_AP
,<chr>,<chr>,<chr>,<chr>,<chr>,<int>,<int>,<lgl>,<dbl>,<dbl>,⋯,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<lgl>
1,2000/01,0002F12C1F92EC01CB0C05D441AA6D7AE16E9CD131BB64885744D08BA74E1FDD,2001-05-15,M,White British,0,0,FALSE,0.66,0.66,⋯,107442,107442,107442,107442,107442,107442,143114,143114,143114,FALSE
2,2000/01,00034E776A98176C2B6075634BC4A614400F32AA9C4D599FDDE156541F1F9A29,2001-01-15,F,White British,0,0,TRUE,3.50,0.62,⋯,140569,140569,140569,140569,140569,140569,140569,140569,140569,FALSE
3,2000/01,00149D26484ED3BE2DEF154B8DB53A36B1A9FF0839C447A2CE19B0750225BA66,2000-11-15,M,South Asian,1,0,FALSE,1.69,0.44,⋯,108087,108087,108087,108087,108087,108087,108087,108087,108087,FALSE
4,2000/01,0014DF7EADF7C22A0FD2F6862B6F06778B199F1F2D61242125DA8EC539E6275D,2001-04-15,F,White British,0,0,FALSE,1.12,0.62,⋯,107366,107366,107366,107366,107366,107366,142761,142761,142761,FALSE
5,2000/01,001B7CEB7B4CC529920A0B9F5B83D1AC0019E5DC44B3FBB3CEF665C331C0C3B6,2001-05-15,M,White British,0,0,FALSE,1.56,1.56,⋯,107395,107395,107395,107395,107395,107395,107395,107395,107395,FALSE
6,2000/01,001E539C263734CC355646811C93CF85B0C587D2EBA8B0D79290FDE8A75D6267,2000-10-15,M,White British,0,0,FALSE,1.79,1.56,⋯,107439,107439,107439,107439,107439,107439,107439,107439,107439,FALSE


In [6]:
cohort2 <- raw_data |>
    filter(cohort == '2001/02') |>
    select(-c(Y2012.2013_Term1, Y2012.2013_Term2, Y2012.2013_Term3)) |>
    rename(
        Year7_Term1 = Y2013.2014_Term1,
        Year7_Term2 = Y2013.2014_Term2,
        Year7_Term3 = Y2013.2014_Term3,
        Year8_Term1 = Y2014.2015_Term1,
        Year8_Term2 = Y2014.2015_Term2,
        Year8_Term3 = Y2014.2015_Term3,
        Year9_Term1 = Y2015.2016_Term1,
        Year9_Term2 = Y2015.2016_Term2,
        Year9_Term3 = Y2015.2016_Term3,
        Year10_Term1 = Y2016.2017_Term1,
        Year10_Term2 = Y2016.2017_Term2,
        Year10_Term3 = Y2016.2017_Term3,
        Year11_Term1 = Y2017.2018_Term1,
        Year11_Term2 = Y2017.2018_Term2,
        Year11_Term3 = Y2017.2018_Term3
        )

In [7]:
head(cohort2)

,cohort,person_id,DOB,Gender,Ethnicity,EAL,ever_FSM,ever_CIN,DistCurrSch,DistNearSch,⋯,Year9_Term1,Year9_Term2,Year9_Term3,Year10_Term1,Year10_Term2,Year10_Term3,Year11_Term1,Year11_Term2,Year11_Term3,ever_AP
,<chr>,<chr>,<chr>,<chr>,<chr>,<int>,<int>,<lgl>,<dbl>,<dbl>,⋯,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<lgl>
1,2001/02,0004E60FEFC16781A48973C80F782B233869629C9CE44B741E2D2043047FCB1D,2001-11-15,M,White British,0,1,TRUE,7.01,0.00,⋯,136529,136529,136529,136529,136529,136529,136529,136529,136529,FALSE
2,2001/02,0013020C6ED2FF78BCA8E5F2D0B9B16E66C50B6A2AABB754FB63314CFE3A2EC7,2002-05-15,M,Other,1,0,FALSE,0.57,0.57,⋯,107441,107441,107441,143112,143112,143112,143112,143112,143112,FALSE
3,2001/02,0018E26A40A6299391164D4918658B0B1248367C762DCE6D2FDF1AE131A9138C,2002-07-15,M,South Asian,1,1,FALSE,0.15,0.15,⋯,139474,139474,139474,139474,139474,139474,139474,139474,139474,FALSE
4,2001/02,0019031CD62586D59268772800516ED50CDF1EF7DE46488542BB031850F6E68A,2001-12-15,F,Other,1,1,TRUE,0.57,0.57,⋯,139982,139982,139982,139982,139982,139982,139982,139982,139982,FALSE
5,2001/02,001E5C1FFA964716BF5E84562BC0B5D64F7742B1E8126718A5C0157D485F215D,2002-04-15,M,White British,0,0,FALSE,3.60,3.60,⋯,136905,136905,136905,136905,136905,136905,136905,136905,136905,FALSE
6,2001/02,001F566795F57F5D8DD92430D868A38B757699844E3E09569B33253527012932,2002-06-15,M,White British,0,0,FALSE,0.29,0.29,⋯,107440,107440,107440,107440,107440,107440,107440,107440,107440,FALSE


In [8]:
data <- rbind(cohort1,cohort2) 

In [9]:
data

cohort,person_id,DOB,Gender,Ethnicity,EAL,ever_FSM,ever_CIN,DistCurrSch,DistNearSch,⋯,Year9_Term1,Year9_Term2,Year9_Term3,Year10_Term1,Year10_Term2,Year10_Term3,Year11_Term1,Year11_Term2,Year11_Term3,ever_AP
<chr>,<chr>,<chr>,<chr>,<chr>,<int>,<int>,<lgl>,<dbl>,<dbl>,⋯,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<lgl>
2000/01,0002F12C1F92EC01CB0C05D441AA6D7AE16E9CD131BB64885744D08BA74E1FDD,2001-05-15,M,White British,0,0,FALSE,0.66,0.66,⋯,107442,107442,107442,107442,107442,107442,143114,143114,143114,FALSE
2000/01,00034E776A98176C2B6075634BC4A614400F32AA9C4D599FDDE156541F1F9A29,2001-01-15,F,White British,0,0,TRUE,3.50,0.62,⋯,140569,140569,140569,140569,140569,140569,140569,140569,140569,FALSE
2000/01,00149D26484ED3BE2DEF154B8DB53A36B1A9FF0839C447A2CE19B0750225BA66,2000-11-15,M,South Asian,1,0,FALSE,1.69,0.44,⋯,108087,108087,108087,108087,108087,108087,108087,108087,108087,FALSE
2000/01,0014DF7EADF7C22A0FD2F6862B6F06778B199F1F2D61242125DA8EC539E6275D,2001-04-15,F,White British,0,0,FALSE,1.12,0.62,⋯,107366,107366,107366,107366,107366,107366,142761,142761,142761,FALSE
2000/01,001B7CEB7B4CC529920A0B9F5B83D1AC0019E5DC44B3FBB3CEF665C331C0C3B6,2001-05-15,M,White British,0,0,FALSE,1.56,1.56,⋯,107395,107395,107395,107395,107395,107395,107395,107395,107395,FALSE
2000/01,001E539C263734CC355646811C93CF85B0C587D2EBA8B0D79290FDE8A75D6267,2000-10-15,M,White British,0,0,FALSE,1.79,1.56,⋯,107439,107439,107439,107439,107439,107439,107439,107439,107439,FALSE
2000/01,00222373A6082CDBF94DA04F23E7EB4C8775BEE122FE9C29FBF521900A498EDF,2000-11-15,F,South Asian,1,0,FALSE,2.52,0.61,⋯,107391,107391,107391,107391,107391,107391,143113,143113,143113,FALSE
2000/01,002E041C5100C46E33B8A0C0220BDE85A79111C5D0E26742F6278C059930FAF6,2000-12-15,M,South Asian,0,1,FALSE,1.60,0.21,⋯,130909,130909,130909,130909,130909,130909,130909,130909,130909,FALSE
2000/01,003132A725233755576CF96E5AA2B3DF606A618754D4BDDFCDD5483849E7B946,2001-07-15,M,White British,0,1,FALSE,0.33,0.33,⋯,107441,107441,107441,107441,107441,107441,143112,143112,143112,FALSE


# Missing data

In [10]:
data |> summarise(across(everything(), ~ sum(is.na(.))))

cohort,person_id,DOB,Gender,Ethnicity,EAL,ever_FSM,ever_CIN,DistCurrSch,DistNearSch,⋯,Year9_Term1,Year9_Term2,Year9_Term3,Year10_Term1,Year10_Term2,Year10_Term3,Year11_Term1,Year11_Term2,Year11_Term3,ever_AP
<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,⋯,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>
0,0,0,3,42,288,215,0,325,675,⋯,656,656,656,636,636,636,729,729,729,0


In [11]:
colSums(is.na(data))

cohort               person_id                     DOB 
                      0                       0                       0 
                 Gender               Ethnicity                     EAL 
                      3                      42                     288 
               ever_FSM                ever_CIN             DistCurrSch 
                    215                       0                     325 
            DistNearSch             LSOA11_Home                   IDACI 
                    675                     293                      79 
       ever_SEN_support  first_SEN_support_year        SEN_support_need 
                      0                    9798                    9798 
          ever_EHC_plan     first_EHC_plan_year           EHC_plan_need 
                      0                   16607                   16607 
                 no_SEN   AuthorisedAbsenceRate UnauthorisedAbsenceRate 
                      0                       0                       0 
       TotalAbsenceRate             Suspensions      SuspensionSessions 
                      0                       0                       0 
             Exclusions             KS4_Entries       KS4_Passes_Level1 
                      0                     415                     415 
      KS4_Passes_Level2             Destination             Year7_Term1 
                    415                       0                    1029 
            Year7_Term2             Year7_Term3             Year8_Term1 
                   1029                    1029                     802 
            Year8_Term2             Year8_Term3             Year9_Term1 
                    802                     802                     656 
            Year9_Term2             Year9_Term3            Year10_Term1 
                    656                     656                     636 
           Year10_Term2            Year10_Term3            Year11_Term1 
                    636                     636                     729 
           Year11_Term2            Year11_Term3                 ever_AP 
                    729                     729                       0

263 students with all NAs for school 
- 38 labelled DROP
- 17 risky
- 208 steady eet

415 students with no KS4 data 
- 36 labelled DROP
- 144 risky
- 235 steady eet

- N = 17,224
- drop DROP: n = 16,956
- drop all NAs for school: n = 16,731
- drop NA for KS4 entries: n = 16,390
- drop NA for ever FSM: n = 16,285


In [12]:
data_drop_missing <- data |>
    filter(
        Destination != 'DROP',
        !if_all(c(
                Year7_Term1, Year7_Term2, Year7_Term3,
                Year8_Term1, Year8_Term2, Year8_Term3,
                Year9_Term1, Year9_Term2, Year9_Term3,
                Year10_Term1, Year10_Term2, Year10_Term3,
                Year11_Term1, Year11_Term2, Year11_Term3
                ), is.na),
        !is.na(KS4_Entries)
        )

In [13]:
colSums(is.na(data_drop_missing))

cohort               person_id                     DOB 
                      0                       0                       0 
                 Gender               Ethnicity                     EAL 
                      3                      42                       3 
               ever_FSM                ever_CIN             DistCurrSch 
                    105                       0                      21 
            DistNearSch             LSOA11_Home                   IDACI 
                    355                       6                       0 
       ever_SEN_support  first_SEN_support_year        SEN_support_need 
                      0                    9351                    9351 
          ever_EHC_plan     first_EHC_plan_year           EHC_plan_need 
                      0                   15816                   15816 
                 no_SEN   AuthorisedAbsenceRate UnauthorisedAbsenceRate 
                      0                       0                       0 
       TotalAbsenceRate             Suspensions      SuspensionSessions 
                      0                       0                       0 
             Exclusions             KS4_Entries       KS4_Passes_Level1 
                      0                       0                       0 
      KS4_Passes_Level2             Destination             Year7_Term1 
                      0                       0                     673 
            Year7_Term2             Year7_Term3             Year8_Term1 
                    673                     673                     447 
            Year8_Term2             Year8_Term3             Year9_Term1 
                    447                     447                     298 
            Year9_Term2             Year9_Term3            Year10_Term1 
                    298                     298                     230 
           Year10_Term2            Year10_Term3            Year11_Term1 
                    230                     230                     241 
           Year11_Term2            Year11_Term3                 ever_AP 
                    241                     241                       0

#### Replace svchool NAs with 111111

In [14]:
data_drop_missing <- data_drop_missing |>
    mutate(across(c(Year7_Term1, Year7_Term2, Year7_Term3,
                Year8_Term1, Year8_Term2, Year8_Term3,
                Year9_Term1, Year9_Term2, Year9_Term3,
                Year10_Term1, Year10_Term2, Year10_Term3,
                Year11_Term1, Year11_Term2, Year11_Term3),
                ~ replace_na(., 111111)))

In [15]:
#data_drop_missing <- data_drop_missing |>
#    filter(!is.na(ever_FSM))

In [16]:
head(data_drop_missing)

,cohort,person_id,DOB,Gender,Ethnicity,EAL,ever_FSM,ever_CIN,DistCurrSch,DistNearSch,⋯,Year9_Term1,Year9_Term2,Year9_Term3,Year10_Term1,Year10_Term2,Year10_Term3,Year11_Term1,Year11_Term2,Year11_Term3,ever_AP
,<chr>,<chr>,<chr>,<chr>,<chr>,<int>,<int>,<lgl>,<dbl>,<dbl>,⋯,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<lgl>
1,2000/01,0002F12C1F92EC01CB0C05D441AA6D7AE16E9CD131BB64885744D08BA74E1FDD,2001-05-15,M,White British,0,0,FALSE,0.66,0.66,⋯,107442,107442,107442,107442,107442,107442,143114,143114,143114,FALSE
2,2000/01,00034E776A98176C2B6075634BC4A614400F32AA9C4D599FDDE156541F1F9A29,2001-01-15,F,White British,0,0,TRUE,3.50,0.62,⋯,140569,140569,140569,140569,140569,140569,140569,140569,140569,FALSE
3,2000/01,00149D26484ED3BE2DEF154B8DB53A36B1A9FF0839C447A2CE19B0750225BA66,2000-11-15,M,South Asian,1,0,FALSE,1.69,0.44,⋯,108087,108087,108087,108087,108087,108087,108087,108087,108087,FALSE
4,2000/01,0014DF7EADF7C22A0FD2F6862B6F06778B199F1F2D61242125DA8EC539E6275D,2001-04-15,F,White British,0,0,FALSE,1.12,0.62,⋯,107366,107366,107366,107366,107366,107366,142761,142761,142761,FALSE
5,2000/01,001B7CEB7B4CC529920A0B9F5B83D1AC0019E5DC44B3FBB3CEF665C331C0C3B6,2001-05-15,M,White British,0,0,FALSE,1.56,1.56,⋯,107395,107395,107395,107395,107395,107395,107395,107395,107395,FALSE
6,2000/01,001E539C263734CC355646811C93CF85B0C587D2EBA8B0D79290FDE8A75D6267,2000-10-15,M,White British,0,0,FALSE,1.79,1.56,⋯,107439,107439,107439,107439,107439,107439,107439,107439,107439,FALSE


# Simplified dataset

In [17]:
data_simple <- data_drop_missing |>
    mutate(Highest_SEN_level = case_when (
            no_SEN ~ 'No SEN', 
            ever_SEN_support & !ever_EHC_plan ~ 'SEN Support',
            ever_EHC_plan ~ 'EHC Plan'
        ))

In [18]:
data_simple <- data_simple |>
    select(-c(ever_SEN_support,first_SEN_support_year,SEN_support_need,ever_EHC_plan,first_EHC_plan_year,EHC_plan_need,no_SEN, 
              DistNearSch))

In [19]:
head(data_simple)

,cohort,person_id,DOB,Gender,Ethnicity,EAL,ever_FSM,ever_CIN,DistCurrSch,LSOA11_Home,⋯,Year9_Term2,Year9_Term3,Year10_Term1,Year10_Term2,Year10_Term3,Year11_Term1,Year11_Term2,Year11_Term3,ever_AP,Highest_SEN_level
,<chr>,<chr>,<chr>,<chr>,<chr>,<int>,<int>,<lgl>,<dbl>,<chr>,⋯,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<int>,<lgl>,<chr>
1,2000/01,0002F12C1F92EC01CB0C05D441AA6D7AE16E9CD131BB64885744D08BA74E1FDD,2001-05-15,M,White British,0,0,FALSE,0.66,E01010635,⋯,107442,107442,107442,107442,107442,143114,143114,143114,FALSE,SEN Support
2,2000/01,00034E776A98176C2B6075634BC4A614400F32AA9C4D599FDDE156541F1F9A29,2001-01-15,F,White British,0,0,TRUE,3.50,E01010749,⋯,140569,140569,140569,140569,140569,140569,140569,140569,FALSE,SEN Support
3,2000/01,00149D26484ED3BE2DEF154B8DB53A36B1A9FF0839C447A2CE19B0750225BA66,2000-11-15,M,South Asian,1,0,FALSE,1.69,E01010610,⋯,108087,108087,108087,108087,108087,108087,108087,108087,FALSE,No SEN
4,2000/01,0014DF7EADF7C22A0FD2F6862B6F06778B199F1F2D61242125DA8EC539E6275D,2001-04-15,F,White British,0,0,FALSE,1.12,E01010815,⋯,107366,107366,107366,107366,107366,142761,142761,142761,FALSE,SEN Support
5,2000/01,001B7CEB7B4CC529920A0B9F5B83D1AC0019E5DC44B3FBB3CEF665C331C0C3B6,2001-05-15,M,White British,0,0,FALSE,1.56,E01010575,⋯,107395,107395,107395,107395,107395,107395,107395,107395,FALSE,No SEN
6,2000/01,001E539C263734CC355646811C93CF85B0C587D2EBA8B0D79290FDE8A75D6267,2000-10-15,M,White British,0,0,FALSE,1.79,E01010704,⋯,107439,107439,107439,107439,107439,107439,107439,107439,FALSE,SEN Support


# Save CSVs

In [20]:
# save as csv
write.csv(data, "data/data_EA.csv", row.names = FALSE)

In [21]:
# save as csv
write.csv(data_drop_missing, "data/data_EA_noNA.csv", row.names = FALSE)

In [22]:
# save as csv
write.csv(data_simple, "data/data_simple.csv", row.names = FALSE)